cmd/duffle/main_test.go

```diff
@@ -162,3 +164,85 @@ func TestFindCreds(t *testing.T) {
 		os.RemoveAll(filepath.Join(credDir, "*"))
 	}
 }
+
+func TestMakeOpRelocator(t *testing.T) {
+	is := assert.New(t)
+
+	tests := map[string]struct {
+		relMapFile                      string
+		invocationImage                 string
+		expectedErrorMessage            string
+		expectedInvocationImage         string
+		expectedOpRelocatorErrorMessage string
+	}{
+		"bad relocation mapping file": {
+			relMapFile:                      "no-such-file",
+			invocationImage:                 "example.com/original",
+			expectedErrorMessage:            "failed to read relocation mapping from no-such-file:",
+			expectedInvocationImage:         "",
+			expectedOpRelocatorErrorMessage: "",
+		},
+		"valid relocation mapping file": {
+			relMapFile:                      "testdata/oprelocator/relmap.json",
+			invocationImage:                 "example.com/original",
+			expectedErrorMessage:            "",
+			expectedInvocationImage:         "example.com/relocated",
+			expectedOpRelocatorErrorMessage: "",
+		},
+		"omitted relocation mapping file": {
+			relMapFile:                      "",
+			invocationImage:                 "example.com/original",
+			expectedErrorMessage:            "",
+			expectedInvocationImage:         "example.com/original",
+			expectedOpRelocatorErrorMessage: "",
+		},
+		"non-existent relocation mapping file": {
```

Comment:
What is if the difference between this test case and "bad relocation mapping file"? They both seem to have the same setup with a non-existent relocation mapping file, but then they each have different `expectedInvocationImage` which I don't quite understand.

Reply:
Well spotted! There is no significant difference, so I've deleted one of them. `expectedInvocationImage` is unused since `expectedErrorMessage` is non-empty and so its value makes no difference.

Reply:
Pushed a fix and also improved the test to check the mounting of the relocation mapping.